### ISSUE TITLE
AWS Batch Environment INVALID

### ISSUE THREAD

**Issue by Tomczik76** [aws-samples/performance-testing-framework-for-apache-kafka#5]
I'm trying to use the performance test but I'm getting this error in my AWS Batch Environment
```
CLIENT_ERROR - One or more security groups in the launch configuration are not linked to the VPCs configured in the Auto Scaling group
```
I'm trying to use an existing Kafka deployment

```
new CdkStack(app, 'demo-cluster-1', {
  ...defaults,
  vpc: vpc,
  clusterName: 'demo-cluster-1',
  bootstrapBrokerString: 'xxxxx',
  sg: 'sg-xxxxx',
});

```

The security group I provided is the one used by my MSK cluster in my default VPC. If I understand the error correctly the security group isn't in the VPC created by the CDK.

**Comment by sthm** [comment]
You are right. By default, the CDK stack will create a new VPC for the resources that are used to run the test.

You can change that behavior by importing your own VPC into the stack: https://github.com/aws-samples/performance-testing-framework-for-apache-kafka/blob/main/cdk/bin/cdk.ts#L36. In this way all resources reside in the same VPC and therefore the same security group can be shared between the cluster and the instances required for the test.
